core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -91,8 +91,18 @@ protected void waitUntilReady() {
         }
     }
 
-    private boolean isUsingSocketProxyOnMac() {
-        return DockerClientFactory.instance().isUsing(ProxiedUnixSocketClientProviderStrategy.class)
-                && System.getProperty("os.name").toLowerCase().contains("mac");
+    private boolean isApplicable() {
```

Comment:
Maybe this could be named more clearly? Something like `doesDockerPrematurelyListen` or `shouldApplyProxyWorkaround`?